### PR TITLE
Feature/file provider

### PR DESCRIPTION
**Pull request by jamesmoey**
Following my previous PR. This is a file provider. Message are store as file. Using flock to prevent multiple access to the same message. This implement is missing the following feature.

 - Notification feature
 - Message Timeout

### CODE REVIEW

**Comment by jamesmoey** [comment]
@aequasi Any feedback on this?

**Comment by kmfk** [comment]
@jamesmoey - think its waiting on me.  sorry, we were wrapping up a release at work and I've been swamped.  Let me take a look through this and #73 - should have some time this weekend to get everything merged in.

File: composer.json [hunk 1]
```diff
@@ -19,17 +19,24 @@
     "require": {
         "php":                      ">=5.4.0",
         "doctrine/common":          "~2.4",
-        "symfony/framework-bundle": "~2.3",
-        "symfony/monolog-bundle":   "~2.3"
+        "symfony/dependency-injection": "~2.3, <3.0.0@dev",
```
Comment (xabbuh) [inline composer.json]: `~2.3` is sufficient. You won't get `3.0` with it anyway.

File: src/Command/QueueBuildCommand.php [hunk 2]
```diff
@@ -22,16 +22,20 @@
...
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
+use Symfony\Component\DependencyInjection\ContainerAwareTrait;
 
 /**
  * @author Keith Kirk <kkirk@undergroundelephant.com>
  */
-class QueueBuildCommand extends ContainerAwareCommand
+class QueueBuildCommand extends Command implements ContainerAwareInterface
 {
+    use ContainerAwareTrait;
```
Comment (xabbuh) [inline src/Command/QueueBuildCommand.php]: The `ContainerAwareTrait` was introduced in Symfony 2.4.

Comment (jamesmoey) [inline src/Command/QueueBuildCommand.php]: It is removed.

**Comment by kmfk** [comment]
@jamesmoey - Look good to me

**Pull request merged**